packages/docmanager/src/manager.ts

```diff
@@ -126,6 +126,17 @@ export class DocumentManager implements IDocumentManager {
     });
   }
 
+  /**
+   * Whether to prompt to name file on first save.
+   */
+  get nameFileOnSave(): boolean {
+    return this._nameFileOnSave;
+  }
+
+  set nameFileOnSave(value: boolean) {
```

Comment:
Is this stored per document or per global setting. It feels like it should be a setting?

Reply:
@goanpeca you mentioned Signal to settingRegister on change in nameFileOnSave. I am actually not too sure how that work. any pointer to an example maybe?

Reply:
See this line, https://github.com/jupyterlab/jupyterlab/pull/10043/files#diff-e0f9ab316153ea5eac9b4e2757412d93bbd75b744c89e8c88818f7ce795bd2d7L42

We could add a new signal to the interface. And make make it more generic like `optionChanged` ?

`readonly optionChanged: ISignal<this, Object>;`

then on the set method

```ts
set nameFileOnSave(value: boolean) {
    this._nameFileOnSave = value;
    this.optionChanged.emit({nameFileOnSave: value});
}
```


And then on the docmanager-extension file we could do something like:


```ts
    docManager.optionChanged.connect(...)
```

Does this make sense?

Could you also look at how other components implement `IOptions` that way we can follow the same paradigm for the constructor?

Reply:
@goanpeca thx for the info! I tired it out and looks like using Signal has a lot of potentials. and Yes thanks for suggesting `IOptions`.